lib/fhir/resources/practitioner.js

```diff
@@ -65,39 +65,7 @@ module.exports = (mongo) => {
           return callback(null, fhirCommon.buildHTTPOutcome(400, 'error', 'invalid', badRequest.message))
         }
 
-        let query = { $and: [] }
-
-        if (queryObject['identifier']) {
-          query['$and'].push(queryUtils.tokenToSystemValueElemMatch('identifier', queryObject['identifier'][constants.NO_MODIFER]))
-        }
-
-        if (queryObject['given']) {
-          query['$and'].push(queryUtils.nameToMongoClause('given', queryObject['given'][constants.NO_MODIFER]))
-        }
-
-        if (queryObject['family']) {
-          query['$and'].push(queryUtils.nameToMongoClause('family', queryObject['family'][constants.NO_MODIFER]))
-        }
-
-        if (queryObject['role']) {
-          query['$and'].push(queryUtils.tokenToSystemCodeElemMatch('practitionerRole.role.coding', queryObject['role'][constants.NO_MODIFER]))
-        }
-
-        if (queryObject['organization']) {
-          let clause = {}
-          clause['practitionerRole'] = {
-            $elemMatch: {
-              managingOrganization: {
-                reference: queryUtils.paramAsReference(queryObject['organization'][constants.NO_MODIFER], 'Organization')
-              }
-            }
-          }
-          query['$and'].push(clause)
-        }
-
-        if (queryObject['telecom']) {
-          query['$and'].push(queryUtils.tokenToSystemValueElemMatch('telecom', queryObject['telecom'][constants.NO_MODIFER]))
-        }
+        const query = queryBuilder.buildQuery(ctx.resourceType, queryObject)
```

Comment:
Lets add in the customParams filter here for completeness, There are no custom params at the moment, but possible in future
`const filteredQueryObject = queryUtils.filterOutProperties(queryObject, Object.keys(customSupportedParams))`